prow/git/git.go

```diff
@@ -288,9 +296,17 @@ func (r *Repo) Push(repo, branch string) error {
 		return errors.New("cannot push without credentials - configure your git client")
 	}
 	r.logger.Infof("Pushing to '%s/%s (branch: %s)'.", r.user, repo, branch)
-	remote := fmt.Sprintf("https://%s:%s@%s/%s/%s", r.user, r.pass, github, r.user, repo)
+
+	gitURL, err := url.Parse(r.base)
+	if err != nil {
+		return err
+	}
+	gitURL.User = url.UserPassword(r.user, r.pass)
+	gitURL.Path = strings.Join([]string{r.user, repo}, "/")
```

Comment:
Sorry guys, this should be the last change... I found that this was removed during reuse of the `u` variable, so wanted to get this in so we didn't have issues.

Please review and let me know if we need anything else before merge

Reply:
Did this fail any tests when it was missing? If not, can we add a test that would have failed?

Reply:
Can we make this a future enhancement? There isn't any existing testing for `git.Push` and the function seems it would require re-work to support mock testing.

Reply:
Sure -- I would suggest making the implementation of `func (r *Repo) gitCommand(arg ...string)` use a member of `type Repo struct` and injecting that during test time to record the actions that would be taken, that would allow us to test the underlying git call for input args to the client from Go.

Reply:
Ideally, unit tests to validate new functionality should come in the same PR as the new functionality. We should consider this a future enhancement.

My concern is that we LGTM'd (and therefore almost merged) something which would have caused an outage and no test failed to catch that scenario.

This suggests we are missing useful test coverage, an easy way to test this would be to encapsulate `gitURL(r Repo) (string, error)` and then do
```golang
cases := []struct{
  name string
  input Repo
  expected string
  err bool
}{
   { name: "no user returns error",
     err: true,
   },
  { // etc
  }
}

for _, tc := range cases {
  actual, err := gitRepo(tc.Repo)
  if actual != tc.expected { tc.Errorf() }
}
```

Maybe gitRemote is a better name?

Then this function becomes pretty trivial:
```golang
remote, err := gitRemote(repo)
return r.gitCommand("push", remote, branch).CombinedOutput()
```

which I don't think requires further unit testing